### DISCUSSION TITLE
Can skywalking display the metrics(such as resp_time or cpm)of a logic endpoint, in the service dashboard (not endpoint dashboard)?

### DISCUSSION THREAD

**Discussion by willardwang** [apache/skywalking#10864] [Q&A] [answered]
hi，I customized the metric in the config/oal/core.oal, specifying the name of the logic endpoint, but after adding this metric to the service dashboard, metric is empty。thx a lot

`service_aws_form_resp_time = from(Endpoint.latency).filter(name == "customFormPage").longAvg();`

<img width="1786" alt="image" src="https://github.com/apache/skywalking/assets/3304388/fb26c2cd-603c-4545-9dcb-3683e1949b31">

**Comment by wu-sheng** [comment]
How is it possible? I can't follow your point. The context as least requires an endpoint name to read metric.

**Reply by willardwang** [reply]
Thanks for your reply, I was wondering if I could specify an endpoint name in the mal expression，For example, using filter。
service_aws_form_resp_time = from(Endpoint.latency)**.filter(name == "customFormPage")**.longAvg();

**Reply by wu-sheng** [reply]
Then it is still an endpoint, nothing changed. If you don't give a name, how to index that metric?

**Reply by willardwang** [reply]
The background of the discussion is that I use logic endpoint to record metrics for a method, and I want to show the metrics about this method to be displayed on the service dashboard

**Reply by willardwang** [reply]
Thanks for your reply, I need to further study the mechanics of the skywalking metric

**Reply by wu-sheng** [reply, marked as answer]
Then it is not a logic endpoint. You could try to use tag equal to build a metric of service. 
Logic endpoint is about building an endpoint level thing to have all metrics of other default endpoints.

**Reply by willardwang** [reply]
thx a lot!
